codec-http/src/main/java/io/netty/handler/codec/http/multipart/HttpData.java

```diff
@@ -99,6 +99,23 @@
     long length();
 
     /**
+     * Returns the current defined length of the HttpData.
+     *
+     * If no Content-Length is provided in the request, this is equivalent to the current size (length()) since the
+     * decoding does not known the final size yet. Therefore the defined length is
+     * always 0 the current length (whatever during decoding or in final state).
```

Comment:
When no `Content-Length` is provided, the attribute/fileUpload has no information on final size so defined size is 0 and never changes.